### ISSUE TITLE
[bug] When resizable is false window size is forced to be at least 200px (Linux)

### ISSUE THREAD

**Issue by Megamannen** [tauri-apps/tao#561]
Using this sample
[resizable](https://github.com/tauri-apps/tao/blob/dev/examples/resizable.rs)

When starting with
```rs
  let mut resizable = true;

  let window = WindowBuilder::new()
    .with_title("Hit space to toggle resizability.")
    .with_inner_size(LogicalSize::new(220.0, 90.0))
    .with_resizable(resizable)
    .build(&event_loop)
    .unwrap();
```
It starts out at the correct size, but when pressing space (triggering resizable change), window size is forced to 200px height

**Comment by Megamannen** [comment]
[tao-resizeable.zip](https://github.com/tauri-apps/tao/files/9613582/tao-resizeable.zip)

Or sample project with the same code as the sample

**Comment by amrbashir** [comment]
cc @wusyong

**Comment by amrbashir** [comment]
Just tested this on my KDE and it works fine, what is your Linux distro and Window manager?

**Comment by Megamannen** [comment]
> Just tested this on my KDE and it works fine, what is your Linux distro and Window manager?

Manjaro, running Gnome Desktop. Can't give you exact version number right now since I'm not at work, but I'm 99% certain it's Gnome 43

**Comment by amrbashir** [comment]
@FabianLars ☝️

**Comment by FabianLars** [comment]
yep, same issue using the zip from above on ubuntu2204 on wsl, and fedora+gnome (both wayland and x11)

**Comment by JakubGluszek** [comment]
Same issue on Ubuntu 22.10 (5.19.0-35-generic) (X11 / i3)

**Comment by JakubGluszek** [comment]
> Same issue on Ubuntu 22.10 (5.19.0-35-generic) (X11 / i3)

I also wanted to add that the issue is present only when the window is in "floating" mode (when using i3 WM).

**Comment by yuezk** [comment]
Same issue here on Ubuntu 22.04, GNOME version: 42.5, Wayland

**Comment by Daniel-Ros** [comment]
Same issue on opensuse, hyprland/wayland

**Comment by clemux** [comment]
Same issue here, both with GNOME 44.2 on Wayland and with i3 on X11. 
(webkit2gtk-4.0: 2.40.4, tauri: 1.4.1)

**Comment by FoliageOwO** [comment]
Same issue, running on `Manjaro Linux` with `Plasma 5.27.10` as DE and `KWin` as WM. When will it be fixed? :worried:

**Comment by imSarahOwl** [comment]
Hey, are there any updates on this issue?
I'm having the same issue on `Arch Linux`, with `Plasma 6.3.2`, also on `Hyprland 0.47.2`. I also tried on an `Ubuntu 22.04.2` vm with `Gnome 42.9` to sanity check myself

**Comment by FabianLars** [comment]
no updates from our side

**Comment by imSarahOwl** [comment]
alright, ty!!

**Comment by l-mendes** [comment]
I was facing the same issue (Fedora 41 - GNOME Shell 47.10 - Wayland) and resolved it by following the instructions below:

- Added the GTK and WebKitGTK dependencies to `Cargo.toml` for the Linux platform:

```toml
[target.'cfg(target_os = "linux")'.dependencies]
gtk = "0.18"
webkit2gtk = "2.0"
```

- Configured the window size directly in Rust `src-tauri/src/lib.rs`, inside the Tauri configuration block:

```rust
  tauri::Builder::default()
    // ... builder code
    .setup(|app| {
      // ... setup code
      #[cfg(target_os = "linux")]
      if let Some(tray_win) = app.get_webview_window("tray") {
          use gtk::prelude::WidgetExt;
          let _ = tray_win.with_webview(|webview| {
              webview.inner().set_size_request(300, 48);
          });
      }
      // ... other setup code
      Ok(())
    })
    // ... other parts of the builder code
```

This ensured the window had the correct dimensions, despite WebKitGTK's peculiarities on Linux.
